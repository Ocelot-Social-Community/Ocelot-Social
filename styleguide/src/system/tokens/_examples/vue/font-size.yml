#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
