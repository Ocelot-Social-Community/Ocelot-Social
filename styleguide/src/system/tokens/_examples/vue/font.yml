#
# FONT TOKENS
# Use these tokens to set font-weight and font-family.
#

props:
  - name: font-family-heading
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-text
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-code
    value: "inconsolata, monospace"
    category: "font-family"

  - name: font-weight-regular
    value: "normal"
    category: "font-weight"
  - name: font-weight-bold
    value: "600"
    category: "font-weight"
global:
  type: ...
  category: font
