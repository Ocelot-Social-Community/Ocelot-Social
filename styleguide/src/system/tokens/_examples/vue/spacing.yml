#
# SPACING TOKENS
# Use these tokens to set padding, margin and position coordinates.
#

aliases:
  xxx-large:
    value: &xxx-large 120
  xx-large:
    value: &xx-large 60
  x-large:
    value: &x-large 40
  large:
    value: &large 28
  base:
    value: &base 20
  small:
    value: &small 12
  x-small:
    value: &x-small 6
  xx-small:
    value: &xx-small 4
  xxx-small:
    value: &xxx-small 2

props:
  - name: xxx-large
    value: *xxx-large
    category: space-size
  - name: xx-large
    value: *xx-large
    category: space-size
  - name: x-large
    value: *x-large
    category: space-size
  - name: large
    value: *large
    category: space-size
  - name: base
    value: *base
    category: space-size
  - name: small
    value: *small
    category: space-size
  - name: x-small
    value: *x-small
    category: space-size
  - name: xx-small
    value: *xx-small
    category: space-size
  - name: xxx-small
    value: *xxx-small
    category: space-size

  - name: space-xxx-large
    value: "{!xxx-large}px"
  - name: space-xx-large
    value: "{!xx-large}px"
  - name: space-x-large
    value: "{!x-large}px"
  - name: space-large
    value: "{!large}px"
  - name: space-base
    value: "{!base}px"
  - name: space-small
    value: "{!small}px"
  - name: space-x-small
    value: "{!x-small}px"
  - name: space-xx-small
    value: "{!xx-small}px"
  - name: space-xxx-small
    value: "{!xxx-small}px"

global:
  type: number
  category: space
