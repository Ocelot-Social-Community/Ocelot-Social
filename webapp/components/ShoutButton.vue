<template>
  <ds-space margin="xx-small" class="text-align-center">
    <base-button
      :loading="loading"
      :disabled="disabled"
      :filled="shouted"
      icon="heart-o"
      circle
      @click="toggle"
    />
    <ds-space margin-bottom="xx-small" />
    <ds-text color="soft" class="shout-button-text">
      <ds-heading style="display: inline" tag="h3">{{ shoutedCount }}x</ds-heading>
      {{ $t('shoutButton.shouted') }}
    </ds-text>
  </ds-space>
</template>

<script>
import gql from 'graphql-tag'

export default {
  props: {
    count: { type: Number, default: 0 },
    postId: { type: String, default: null },
    isShouted: { type: Boolean, default: false },
    disabled: { type: Boolean, default: false },
  },
  data() {
    return {
      loading: false,
      shoutedCount: this.count,
      shouted: false,
    }
  },
  watch: {
    isShouted: {
      immediate: true,
      handler: function (shouted) {
        this.shouted = shouted
      },
    },
  },
  methods: {
    toggle() {
      const shout = !this.shouted
      const mutation = shout ? 'shout' : 'unshout'
      const count = shout ? this.shoutedCount + 1 : this.shoutedCount - 1

      const backup = {
        shoutedCount: this.shoutedCount,
        shouted: this.shouted,
      }

      this.shoutedCount = count
      this.shouted = shout

      this.$apollo
        .mutate({
          mutation: gql`
            mutation($id: ID!) {
              ${mutation}(id: $id, type: Post)
            }
          `,
          variables: {
            id: this.postId,
          },
        })
        .then((res) => {
          if (res && res.data) {
            this.$emit('update', shout)
          }
        })
        .catch(() => {
          this.shoutedCount = backup.shoutedCount
          this.shouted = backup.shouted
        })
        .finally(() => {
          this.loading = false
        })
    },
  },
}
</script>

<style lang="scss">
.shout-button-text {
  user-select: none;
}
.text-align-center {
  text-align: center;
}
</style>
