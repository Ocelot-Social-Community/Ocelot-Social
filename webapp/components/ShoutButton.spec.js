import { render, screen, fireEvent } from '@testing-library/vue'
import '@testing-library/jest-dom'
import Vue from 'vue'
import ShoutButton from './ShoutButton.vue'

const localVue = global.localVue

describe('ShoutButton.vue', () => {
  let mocks

  beforeEach(() => {
    mocks = {
      $t: jest.fn((t) => t),
      $apollo: {
        mutate: jest.fn(),
      },
    }
  })

  describe('mount', () => {
    let wrapper
    const Wrapper = () => {
      return render(ShoutButton, { mocks, localVue })
    }

    beforeEach(() => {
      wrapper = Wrapper()
    })

    it('renders button and text', () => {
      expect(wrapper.container).toMatchSnapshot()
      const button = screen.getByRole('button')
      expect(button).toBeInTheDocument()
    })

    it('toggle the button', async () => {
      mocks.$apollo.mutate = jest.fn().mockResolvedValue({ data: { shout: 'WeDoShout' } })
      const button = screen.getByRole('button')
      await fireEvent.click(button)
      expect(wrapper.container).toMatchSnapshot()
      const shoutedCount = screen.getByText('1')
      expect(shoutedCount).toBeInTheDocument()
    })

    it('toggle the button, but backend fails', async () => {
      mocks.$apollo.mutate = jest.fn().mockRejectedValue({ message: 'Ouch!' })
      const button = screen.getByRole('button')
      await fireEvent.click(button)
      expect(wrapper.container).toMatchSnapshot()
      let shoutedCount = screen.getByText('1')
      expect(shoutedCount).toBeInTheDocument()
      await Vue.nextTick()
      shoutedCount = screen.getByText('0')
      expect(shoutedCount).toBeInTheDocument()
    })
  })
})
