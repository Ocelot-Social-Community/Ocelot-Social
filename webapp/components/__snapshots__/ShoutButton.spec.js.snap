// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoutButton.vue renders button and text 1`] = `
<div>
  <div
    class="action-button"
  >
    <button
      aria-label="shoutButton.shouted"
      class="base-button --icon-only --circle"
      type="button"
    >
      <span
        class="base-icon"
      >
        <!---->
      </span>
       
      <!---->
       
    </button>
     
    <div
      class="count"
    >
      0
    </div>
  </div>
</div>
`;

exports[`ShoutButton.vue toggle the button 1`] = `
<div>
  <div
    class="action-button"
  >
    <button
      aria-label="shoutButton.shouted"
      class="base-button --icon-only --circle --filled"
      type="button"
    >
      <span
        class="base-icon"
      >
        <!---->
      </span>
       
      <!---->
       
    </button>
     
    <div
      class="count"
    >
      1
    </div>
  </div>
</div>
`;

exports[`ShoutButton.vue toggle the button, but backend fails 1`] = `
<div>
  <div
    class="action-button"
  >
    <button
      aria-label="shoutButton.shouted"
      class="base-button --icon-only --circle --filled"
      type="button"
    >
      <span
        class="base-icon"
      >
        <!---->
      </span>
       
      <!---->
       
    </button>
     
    <div
      class="count"
    >
      1
    </div>
  </div>
</div>
`;

exports[`ShoutButton.vue when shouted renders 1`] = `
<div>
  <div
    class="action-button"
  >
    <button
      aria-label="shoutButton.shouted"
      class="base-button --icon-only --circle --filled"
      type="button"
    >
      <span
        class="base-icon"
      >
        <!---->
      </span>
       
      <!---->
       
    </button>
     
    <div
      class="count"
    >
      0
    </div>
  </div>
</div>
`;
